Creating a new note crashes the app

<!-- Checklist for reporting a bug
- Check the issue tracker. The issue you have may have already been reported.
- Ensure you have tested the latest developers version.
-->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
The app crashes when I open a book and click on **New Note**.

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
The app should not crash and **New Note** should be renamed to **Notes** and moved down the menu.

**Steps to reproduce the behavior:**
1. Open a book.
1. Go to menu -> New Note
<!--
2. Click on '....'
3. Scroll down to '....'
4. See error 
-->

**Environment**
- Version of Kiwix Android : develop
- Device : Redmi Note 3
- OS version : Android M

**Logs**
```
java.lang.StringIndexOutOfBoundsException: length=110; regionStart=43; regionLength=-8
        at java.lang.String.startEndAndLength(String.java:298)
        at java.lang.String.substring(String.java:1087)
        at org.kiwix.kiwixmobile.main.AddNoteDialog.getTextAfterLastSlashWithoutExtension(AddNoteDialog.java:205)
        at org.kiwix.kiwixmobile.main.AddNoteDialog.getArticleNotefileName(AddNoteDialog.java:188)
        at org.kiwix.kiwixmobile.main.AddNoteDialog.onCreate(AddNoteDialog.java:105)
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2414)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1418)
```

FYI I reproduced this crash using Android Monkey which triggered the crash after a long stream of events.

```
E/kiwix   (10635): Exception reading article A/index.html from zim file
E/kiwix   (10635): java.io.IOException: write failed: EPIPE (Broken pipe)
E/kiwix   (10635):      at libcore.io.IoBridge.write(IoBridge.java:455)
E/kiwix   (10635):      at java.io.FileOutputStream.write(FileOutputStream.java:187)
E/kiwix   (10635):      at org.kiwix.kiwixmobile.data.ZimContentProvider$TransferThread.run(ZimContentProvider.java:486)
E/kiwix   (10635): Caused by: libcore.io.ErrnoException: write failed: EPIPE (Broken pipe)
E/kiwix   (10635):      at libcore.io.Posix.writeBytes(Native Method)
E/kiwix   (10635):      at libcore.io.Posix.write(Posix.java:202)
E/kiwix   (10635):      at libcore.io.BlockGuardOs.write(BlockGuardOs.java:197)
E/kiwix   (10635):      at libcore.io.IoBridge.write(IoBridge.java:450)
E/kiwix   (10635):      ... 2 more
D/kiwix   (10635): invalidUrl = false
D/kiwix   (10635): Loaded URL: null
W/AwContents(10635): nativeOnDraw failed; clearing to background color.
D/kiwix   (10635): Loaded URL: null
D/kiwix   (10635): invalidUrl = false

... lots of lines cut ...
E/ActivityManager(  821): App crashed! Process: org.kiwix.kiwixcustomphet
E/AndroidRuntime(10635): FATAL EXCEPTION: main
E/AndroidRuntime(10635): Process: org.kiwix.kiwixcustomphet, PID: 10635
E/AndroidRuntime(10635): java.lang.NullPointerException
E/AndroidRuntime(10635):        at org.kiwix.kiwixmobile.main.AddNoteDialog.getTextAfterLastSlashWithoutExtension(AddNoteDialog.java:201)
E/AndroidRuntime(10635):        at org.kiwix.kiwixmobile.main.AddNoteDialog.getArticleNotefileName(AddNoteDialog.java:188)
E/AndroidRuntime(10635):        at org.kiwix.kiwixmobile.main.AddNoteDialog.onCreate(AddNoteDialog.java:105)
E/AndroidRuntime(10635):        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2414)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1418)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
E/AndroidRuntime(10635):        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:802)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManager.java:2625)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2411)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2366)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2273)
E/AndroidRuntime(10635):        at androidx.fragment.app.FragmentManagerImpl$1.run(FragmentManager.java:733)
E/AndroidRuntime(10635):        at android.os.Handler.handleCallback(Handler.java:733)
E/AndroidRuntime(10635):        at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(10635):        at android.os.Looper.loop(Looper.java:157)
E/AndroidRuntime(10635):        at android.app.ActivityThread.main(ActivityThread.java:5633)
E/AndroidRuntime(10635):        at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(10635):        at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(10635):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:896)
E/AndroidRuntime(10635):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:712)
E/AndroidRuntime(10635):        at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager(  821):   Force finishing activity org.kiwix.kiwixcustomphet/org.kiwix.kiwixmobile.main.MainActivity
D/PMS     (  821): acquireWL(426d0950): PARTIAL_WAKE_LOCK  ActivityManager-Launch 0x1 821 1000 null
```
My guess is that for whatever reason, the article was effectively `null` and therefore the call `String notefileName = getTextAfterLastSlashWithoutExtension(articleUrl);` passes an articleUrl of null which causes `int rightmostSlash = path.lastIndexOf('/');` to crash with the NPE

--- a/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
@@ -20,7 +20,6 @@ import android.view.inputmethod.InputMethodManager;
 import android.widget.EditText;
 import android.widget.TextView;
 import android.widget.Toast;
-
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.appcompat.widget.Toolbar;
@@ -28,20 +27,17 @@ import androidx.core.content.ContextCompat;
 import androidx.core.content.FileProvider;
 import androidx.fragment.app.DialogFragment;
 import androidx.fragment.app.Fragment;
-
-import org.kiwix.kiwixmobile.BuildConfig;
-import org.kiwix.kiwixmobile.R;
-import org.kiwix.kiwixmobile.data.ZimContentProvider;
-import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
-
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
-
-import butterknife.BindView;
-import butterknife.ButterKnife;
-import butterknife.Unbinder;
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.data.ZimContentProvider;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
 
 import static org.kiwix.kiwixmobile.utils.Constants.NOTES_DIRECTORY;
 
@@ -55,7 +51,7 @@ import static org.kiwix.kiwixmobile.utils.Constants.NOTES_DIRECTORY;
  */
 
 public class AddNoteDialog extends DialogFragment
-    implements ConfirmationAlertDialogFragment.UserClickListener {
+  implements ConfirmationAlertDialogFragment.UserClickListener {
 
   public static final String TAG = "AddNoteDialog";
 
@@ -91,8 +87,8 @@ public class AddNoteDialog extends DialogFragment
     super.onCreate(savedInstanceState);
 
     setStyle(DialogFragment.STYLE_NORMAL,
-        sharedPreferenceUtil.nightMode() ? R.style.AddNoteDialogStyle_Night
-            : R.style.AddNoteDialogStyle);
+      sharedPreferenceUtil.nightMode() ? R.style.AddNoteDialogStyle_Night
+        : R.style.AddNoteDialogStyle);
 
     // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
     zimFileName = ZimContentProvider.getZimFile();
@@ -118,7 +114,7 @@ public class AddNoteDialog extends DialogFragment
 
   @Override
   public @NonNull View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
-      @Nullable Bundle savedInstanceState) {
+    @Nullable Bundle savedInstanceState) {
     super.onCreateView(inflater, container, savedInstanceState);
     View view = inflater.inflate(R.layout.dialog_add_note, container, false);
     unbinder = ButterKnife.bind(this, view);
@@ -212,7 +208,8 @@ public class AddNoteDialog extends DialogFragment
     int rightmostDot = path.lastIndexOf('.');
 
     if (rightmostSlash > -1 && rightmostDot > -1) {
-      return path.substring(rightmostSlash + 1, (rightmostDot > rightmostSlash) ? rightmostDot : path.length());
+      return path.substring(
+        rightmostSlash + 1, (rightmostDot > rightmostSlash) ? rightmostDot : path.length());
     }
 
     return ""; // If couldn't find a dot and/or slash in the url
@@ -233,14 +230,14 @@ public class AddNoteDialog extends DialogFragment
   void exitAddNoteDialog() {
     if (noteEdited) {
       Fragment previousInstance = getActivity().getSupportFragmentManager()
-          .findFragmentByTag(ConfirmationAlertDialogFragment.TAG);
+        .findFragmentByTag(ConfirmationAlertDialogFragment.TAG);
 
       if (previousInstance == null) {
         // Custom AlertDialog for taking user confirmation before closing note dialog in case of unsaved changes
         DialogFragment newFragment = new ConfirmationAlertDialogFragment(sharedPreferenceUtil, TAG,
-            R.string.confirmation_alert_dialog_message);
+          R.string.confirmation_alert_dialog_message);
         newFragment.show(getActivity().getSupportFragmentManager(),
-            ConfirmationAlertDialogFragment.TAG);
+          ConfirmationAlertDialogFragment.TAG);
       }
     } else {
       // Closing unedited note dialog straightaway
@@ -294,13 +291,13 @@ public class AddNoteDialog extends DialogFragment
 
   private void showKeyboard() {
     InputMethodManager inputMethodManager =
-        (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+      (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
     inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
   }
 
   void closeKeyboard() {
     InputMethodManager inputMethodManager =
-        (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+      (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
     inputMethodManager.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0);
   }
 
@@ -313,7 +310,7 @@ public class AddNoteDialog extends DialogFragment
     if (isExternalStorageWritable()) {
 
       if (ContextCompat.checkSelfPermission(getContext(),
-          Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+        Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
         Log.d(TAG, "WRITE_EXTERNAL_STORAGE permission not granted");
         showToast(R.string.note_save_unsuccessful, Toast.LENGTH_LONG);
         return;
@@ -392,7 +389,7 @@ public class AddNoteDialog extends DialogFragment
 
     if (noteEdited) {
       saveNote(
-          addNoteEditText.getText().toString()); // Save edited note before sharing the text file
+        addNoteEditText.getText().toString()); // Save edited note before sharing the text file
     }
 
     File noteFile = new File(ZIM_NOTES_DIRECTORY + articleNotefileName + ".txt");
@@ -404,8 +401,8 @@ public class AddNoteDialog extends DialogFragment
         // From Nougat 7 (API 24) access to files is shared temporarily with other apps
         // Need to use FileProvider for the same
         noteFileUri =
-            FileProvider.getUriForFile(getContext(), BuildConfig.APPLICATION_ID + ".fileprovider",
-                noteFile);
+          FileProvider.getUriForFile(getContext(), BuildConfig.APPLICATION_ID + ".fileprovider",
+            noteFile);
       } else {
         noteFileUri = Uri.fromFile(noteFile);
       }
@@ -419,7 +416,7 @@ public class AddNoteDialog extends DialogFragment
       noteFileShareIntent.putExtra(Intent.EXTRA_STREAM, noteFileUri);
       noteFileShareIntent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
       Intent shareChooser = Intent.createChooser(noteFileShareIntent,
-          getString(R.string.note_share_app_chooser_title));
+        getString(R.string.note_share_app_chooser_title));
 
       if (noteFileShareIntent.resolveActivity(getActivity().getPackageManager()) != null) {
         startActivity(shareChooser);

--- a/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
@@ -106,10 +106,14 @@ public class AddNoteDialog extends DialogFragment
 
       ZIM_NOTES_DIRECTORY = NOTES_DIRECTORY + zimNoteDirectoryName + "/";
     } else {
-      showToast(R.string.error_filenotfound, Toast.LENGTH_LONG);
-      closeKeyboard();
-      getFragmentManager().beginTransaction().remove(AddNoteDialog.this).commit();
+      onFailureToCreateAddNoteDialog();
     }
+  }
+
+  private void onFailureToCreateAddNoteDialog() {
+    showToast(R.string.error_filenotfound, Toast.LENGTH_LONG);
+    closeKeyboard();
+    getFragmentManager().beginTransaction().remove(AddNoteDialog.this).commit();
   }
 
   @Override
@@ -185,7 +189,13 @@ public class AddNoteDialog extends DialogFragment
     // Returns url of the form: "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html"
     String articleUrl = ((MainActivity) getActivity()).getCurrentWebView().getUrl();
 
-    String notefileName = getTextAfterLastSlashWithoutExtension(articleUrl);
+    String notefileName = "";
+    if (articleUrl == null) {
+      onFailureToCreateAddNoteDialog();
+      return "Not Available";
+    } else {
+      notefileName = getTextAfterLastSlashWithoutExtension(articleUrl);
+    }
 
     return (!notefileName.isEmpty()) ? notefileName : articleTitle;
   }

--- a/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
@@ -188,7 +188,6 @@ public class AddNoteDialog extends DialogFragment
     String notefileName = "";
     if (articleUrl == null) {
       onFailureToCreateAddNoteDialog();
-      return "Not Available";
     } else {
       notefileName = getTextAfterLastSlashWithoutExtension(articleUrl);
     }

--- a/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
@@ -69,14 +69,12 @@ public class AddNoteDialog extends DialogFragment
   private String zimFileName;
   private String zimFileTitle;
   private String articleTitle;
-  // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
-  private String zimNoteDirectoryName;
   // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
   private String articleNotefileName;
   private boolean noteFileExists = false;
-  boolean noteEdited = false;   // Keeps track of state of the note (whether edited since last save)
+  private boolean noteEdited = false;   // Keeps track of state of the note (whether edited since last save)
 
-  private String ZIM_NOTES_DIRECTORY; // Stores path to directory for the currently open zim's notes
+  private String zimNotesDirectory; // Stores path to directory for the currently open zim's notes
 
   public AddNoteDialog(@NonNull SharedPreferenceUtil sharedPreferenceUtil) {
     this.sharedPreferenceUtil = sharedPreferenceUtil;
@@ -97,10 +95,11 @@ public class AddNoteDialog extends DialogFragment
       zimFileTitle = ZimContentProvider.getZimFileTitle();
       articleTitle = ((MainActivity) getActivity()).getCurrentWebView().getTitle();
 
-      zimNoteDirectoryName = getZimNoteDirectoryName();
+      // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+      String zimNoteDirectoryName = getZimNoteDirectoryName();
       articleNotefileName = getArticleNotefileName();
 
-      ZIM_NOTES_DIRECTORY = NOTES_DIRECTORY + zimNoteDirectoryName + "/";
+      zimNotesDirectory = NOTES_DIRECTORY + zimNoteDirectoryName + "/";
     } else {
       onFailureToCreateAddNoteDialog();
     }
@@ -121,29 +120,22 @@ public class AddNoteDialog extends DialogFragment
 
     toolbar.setTitle(R.string.note);
     toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp);
-    toolbar.setNavigationOnClickListener(new View.OnClickListener() {
-      @Override
-      public void onClick(View v) {
-        closeKeyboard();
-        exitAddNoteDialog();
-      }
+    toolbar.setNavigationOnClickListener(v -> {
+      closeKeyboard();
+      exitAddNoteDialog();
     });
 
-    toolbar.setOnMenuItemClickListener(new Toolbar.OnMenuItemClickListener() {
-      @Override
-      public boolean onMenuItemClick(MenuItem item) {
+    toolbar.setOnMenuItemClickListener(item -> {
+      switch (item.getItemId()) {
+        case R.id.share_note: // Opens app-chooser for sharing the note text file
+          shareNote();
+          break;
 
-        switch (item.getItemId()) {
-          case R.id.share_note: // Opens app-chooser for sharing the note text file
-            shareNote();
-            break;
-
-          case R.id.save_note:  // Saves the note as a text file
-            saveNote(addNoteEditText.getText().toString());
-            break;
-        }
-        return true;
+        case R.id.save_note:  // Saves the note as a text file
+          saveNote(addNoteEditText.getText().toString());
+          break;
       }
+      return true;
     });
 
     toolbar.inflateMenu(R.menu.menu_add_note_dialog);
@@ -226,7 +218,7 @@ public class AddNoteDialog extends DialogFragment
     };
   }
 
-  void exitAddNoteDialog() {
+  private void exitAddNoteDialog() {
     if (noteEdited) {
       Fragment previousInstance = getActivity().getSupportFragmentManager()
         .findFragmentByTag(ConfirmationAlertDialogFragment.TAG);
@@ -257,7 +249,7 @@ public class AddNoteDialog extends DialogFragment
     }
   }
 
-  void enableSaveNoteMenuItem() {
+  private void enableSaveNoteMenuItem() {
     if (toolbar.getMenu() != null) {
       MenuItem saveItem = toolbar.getMenu().findItem(R.id.save_note);
       saveItem.setEnabled(true);
@@ -267,7 +259,7 @@ public class AddNoteDialog extends DialogFragment
     }
   }
 
-  void enableShareNoteMenuItem() {
+  private void enableShareNoteMenuItem() {
     if (toolbar.getMenu() != null) {
       MenuItem shareItem = toolbar.getMenu().findItem(R.id.share_note);
       shareItem.setEnabled(true);
@@ -294,13 +286,13 @@ public class AddNoteDialog extends DialogFragment
     inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
   }
 
-  void closeKeyboard() {
+  private void closeKeyboard() {
     InputMethodManager inputMethodManager =
       (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
     inputMethodManager.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0);
   }
 
-  void saveNote(String noteText) {
+  private void saveNote(String noteText) {
 
     /* String content of the EditText, given by noteText, is saved into the text file given by:
      *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
@@ -315,7 +307,7 @@ public class AddNoteDialog extends DialogFragment
         return;
       }
 
-      File notesFolder = new File(ZIM_NOTES_DIRECTORY);
+      File notesFolder = new File(zimNotesDirectory);
       boolean folderExists = true;
 
       if (!notesFolder.exists()) {
@@ -353,14 +345,14 @@ public class AddNoteDialog extends DialogFragment
      * is displayed in the EditText field (note content area)
      * */
 
-    File noteFile = new File(ZIM_NOTES_DIRECTORY + articleNotefileName + ".txt");
+    File noteFile = new File(zimNotesDirectory + articleNotefileName + ".txt");
 
     if (noteFile.exists()) {
       noteFileExists = true;
 
       StringBuilder contents = new StringBuilder();
       try (BufferedReader input = new BufferedReader(new java.io.FileReader(noteFile))) {
-        String line = null;
+        String line;
 
         while ((line = input.readLine()) != null) {
           contents.append(line);
@@ -379,7 +371,7 @@ public class AddNoteDialog extends DialogFragment
     // No action in case the note file for the currently open article doesn't exist
   }
 
-  void shareNote() {
+  private void shareNote() {
 
     /* The note text file corresponding to the currently open article, given at:
      *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
@@ -391,7 +383,7 @@ public class AddNoteDialog extends DialogFragment
         addNoteEditText.getText().toString()); // Save edited note before sharing the text file
     }
 
-    File noteFile = new File(ZIM_NOTES_DIRECTORY + articleNotefileName + ".txt");
+    File noteFile = new File(zimNotesDirectory + articleNotefileName + ".txt");
 
     Uri noteFileUri = null;
     if (noteFile.exists()) {

--- a/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
@@ -109,7 +109,7 @@ public class AddNoteDialog extends DialogFragment
   private void onFailureToCreateAddNoteDialog() {
     showToast(R.string.error_filenotfound, Toast.LENGTH_LONG);
     closeKeyboard();
-    getFragmentManager().beginTransaction().remove(AddNoteDialog.this).commit();
+    getFragmentManager().beginTransaction().remove(this).commit();
   }
 
   @Override

--- a/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
+++ b/app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java
@@ -202,10 +202,10 @@ public class AddNoteDialog extends DialogFragment
     int rightmostDot = path.lastIndexOf('.');
 
     if (rightmostSlash > -1 && rightmostDot > -1) {
-      return (path.substring(rightmostSlash + 1, rightmostDot));
+      return path.substring(rightmostSlash + 1, (rightmostDot > rightmostSlash) ? rightmostDot : path.length());
     }
 
-    return ""; // If couldn't find the dot and/or slash
+    return ""; // If couldn't find a dot and/or slash in the url
   }
 
   // Override onBackPressed() to respond to user pressing 'Back' button on navigation bar

